src/lib/Liblog/pbs_log.c

```diff
@@ -275,24 +275,16 @@ log_get_tls_data(void)
 int
 log_mutex_lock()
 {
-	struct pbs_log_lock *log_lock = NULL;
-	if ((log_lock = pthread_getspecific(pbs_log_tls_key)) == NULL) {
-		log_lock = calloc(1, sizeof(struct pbs_log_lock));
-		if (!log_lock)
-			return -1;
-
-		if (pthread_setspecific(pbs_log_tls_key, log_lock) != 0) {
-			free(log_lock);
-			return -1;
-		}
-		log_lock->locked = 0;
-	}
+	long log_lock;
```

Comment:
Change this to void *, no need to initialize since we assign it immediately after it is defined. Please don't store pointers in type long, or even unsigned long.

Please remove the definition of struct pbs_log_lock since this is the only place it is used.